[ao][docs] adding info on weight only quantization

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90979

Summary: see https://github.com/pytorch/pytorch/issues/90553,
embedding/weight only quantization was previously not documented well

Test Plan: see docs tests

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao][docs] adding info on weight only quantization

Summary: see https://github.com/pytorch/pytorch/issues/90553,
embedding/weight only quantization was previously not documented well

Test Plan: see docs tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on docs/source/quantization.rst:
```diff
@@ -126,15 +127,14 @@ Note that for FX quantization, the corresponding functionals are also supported.
+|nn.EmbeddingBag            | N                 | Y                  |
 +---------------------------+-------------------+--------------------+
-|nn.Embedding               | Y                 | N                  |
+|nn.Embedding               | N                 | Y                  |
```
should this be "n/a" for static quant?

vkuzo commented on docs/source/quantization.rst:
```diff
@@ -450,7 +458,11 @@ FXPTQ API Example::
   model_to_quantize = copy.deepcopy(model_fp)
   model_to_quantize.eval()
-  qconfig_mapping = QConfigMapping().set_global(torch.quantization.default_dynamic_qconfig)
+  qconfig_mapping = (
```
nit: we don't recommend this anymore because there are a lot of ops (such as activations with output range (0, 1) or (-1, 1) will crash without explicit settings for them.  Should we switch this to the defaault qconfig mapping getter?

That function doesn't include embedding qconfig, ideally we would just add it there and the user doesn't need to care about the default case.

doesn't have to be in this PR!

HDCharles committed:
Update on "[ao][docs] adding info on weight only quantization"


Summary: see https://github.com/pytorch/pytorch/issues/90553,
embedding/weight only quantization was previously not documented well

Test Plan: see docs tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on docs/source/quantization.rst:
```diff
@@ -126,15 +129,14 @@ Note that for FX quantization, the corresponding functionals are also supported.
+|nn.Embedding               | N                 | Y                  |
 +---------------------------+-------------------+--------------------+
-| nn.MultiheadAttention     | Y (through        | Not supported      |
+|nn.MultiheadAttention      | Y (through        | Not supported      |
```
are we using `N` for "not supported"?

jerryzh168 commented on docs/source/quantization.rst:
```diff
@@ -126,15 +129,14 @@ Note that for FX quantization, the corresponding functionals are also supported.
 |                           | custom modules)   |                    |
 +---------------------------+-------------------+--------------------+
-| Activations               | Broadly supported | Un-changed,        |
+|Activations                | Broadly supported | Un-changed,        |
```
same here, should we replace "Un-changed ..." to "N" as well

jerryzh168 commented on docs/source/quantization.rst:
```diff
@@ -800,21 +814,22 @@ Note that for FX Graph Mode Quantization, the corresponding functionals are also
 | | nn.Conv1d/2d/3d         | | Y               | | N                |
 +---------------------------+-------------------+--------------------+
-| | nn.LSTM                 | | N               | | Y                |
+| | nn.LSTM                 | | Y (through      | | Y                |
```
do we have two copies of this table? should we dedup?
